### PR TITLE
[wasm] Wasm.experimentalVersion

### PR DESCRIPTION
**Pull request by commonlisp**
To support websites serving different versions of a WASM binary
(i.e., different binary format versions), we use an experimentalVersion
property to indicate the binary format version supported.

### CODE REVIEW

File: lib/Runtime/Library/JavascriptLibrary.cpp [hunk 1]
```diff
@@ -1117,6 +1117,9 @@ namespace Js
         globalObject->SetPrototype(globalObjectPrototype);
         Recycler * recycler = this->GetRecycler();
 
+#ifdef ENABLE_WASM
+        experimentalVersion = JavascriptNumber::New(WasmLibrary::experimentalVersion, scriptContext);
```
Comment (Cellule) [inline lib/Runtime/Library/JavascriptLibrary.cpp]: Could we put this in `InitializeWasmObject` ?

File: lib/Runtime/Library/JavascriptLibraryBase.h [hunk 2]
```diff
@@ -269,6 +269,7 @@ namespace Js
         Var minValue;
         Var maxValue;
         Var negativeZero;
+        Var experimentalVersion; // Wasm.experimentalVersion
```
Comment (Cellule) [inline lib/Runtime/Library/JavascriptLibraryBase.h]: Put this in `JavascriptLibrary.h` changing `JavascriptLibraryBase` will break api compatibility with Edge which is unnecessary in this case

Comment (commonlisp) [inline lib/Runtime/Library/JavascriptLibraryBase.h]: Got it. Thanks.

**Comment by commonlisp** [comment]
@dotnet-bot test this please

File: lib/Runtime/Library/JavascriptLibrary.h [hunk 3]
```diff
@@ -1236,6 +1236,9 @@ namespace Js
 #if DBG
         void DumpLibraryByteCode();
 #endif
+
+        Var experimentalVersion; // Wasm.experimentalVersion
```
Comment (Cellule) [inline lib/Runtime/Library/JavascriptLibrary.h]: Could you put this next to the `wasmObject` inside the `#ifdef ENABLE_WASM` (line 380) of this file

Comment (Cellule) [inline lib/Runtime/Library/JavascriptLibrary.h]: Actually I was wondering if we needed to keep this as a field in Library. Couldn't it just be a member of the Wasm object and that's it?
What I mean is, do we need to keep a reference to the member in JavascriptLibrary?

Comment (commonlisp) [inline lib/Runtime/Library/JavascriptLibrary.h]: Good point. I dropped that field. Thanks.

**Comment by Cellule** [comment]
LGTM

**Pull request merged**